### PR TITLE
Fixing circular import issue in set_param_value method

### PR DESCRIPTION
**Pull request by andrewlee94**
## Summary/Motivation:
Whilst looking into addressing issues with the enthaply of formation in the Shomate equation formulation, it was found that some of the tests suffered from a circular import issue. This PR fixes this issue.

## Changes proposed in this PR:
- Move the `set_param_value` method to `core.util.misc` and renamed to `set_param_from_config`.
- Add error catching and tests for `set_param_from_config` method.
- Changes units in enthalpy and entropy of formation in RPP3 package.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by andrewlee94** [comment]
@agarciadiego Note that I changed the units of the enthalpy and entropy of formation in the RPP3 package, to make them use the base units defined by the property package rather than the source book. I did this as these parameters are something a user might want to use in a unit model, and I thought it was safer to have these in the default units.

File: idaes/core/util/misc.py [hunk 1]
```diff
@@ -213,6 +215,94 @@ def copy_port_values(destination, source):
...
+                " a value for this parameter and index."
+                .format(b.name, param, index))
+
+    units = param_obj.get_units()
+
+    if isinstance(p_data, tuple):
+        # 11 Dec 2020 - There is currently a bug in Pyomo where trying to
+        # convert the units of a unitless quantity results in a TypeError.
+        # To avoid this, we check here for cases where both the parameter and
+        # value are unitless and just set the value directly.
```
Comment (agarciadiego) [inline idaes/core/util/misc.py:293]: values

File: idaes/generic_models/properties/core/generic/generic_property.py [hunk 2]
```diff
@@ -55,6 +55,9 @@
 
 
 def set_param_value(b, param, units):
+    # We cannot use the standard method in core.util.misc as here the parameter
+    # data is directly attached ot the config block, rather than in a parameter
```
Comment (agarciadiego) [inline idaes/generic_models/properties/core/generic/generic_property.py:59]: attached to

**Review by agarciadiego — COMMENTED** [review]
Other than the small typo LGTM

**Comment by andrewlee94** [comment]
@agarciadiego I fixed the typos you noticed.

**Review by agarciadiego — APPROVED** [review]

File: idaes/core/util/misc.py [hunk 3]
```diff
@@ -213,6 +215,94 @@ def copy_port_values(destination, source):
...
 
 
+def set_param_from_config(b, param, config=None, index=None):
+    """
+    Utility method to set parameter value from a config block. This allows for
+    converting units if required. This method directly sets the value of the
+    parameter.
+
+    Args:
+        b - block on which parameter and config block are defined
```
Comment (jghouse88) [inline idaes/core/util/misc.py:225]: I know this is just an argument for the method but `b` can be block to make it clear.

Comment (jghouse88) [reply idaes/core/util/misc.py:225]: What if `b` being passed is not an instance of `block`?

**Review by jghouse88 — APPROVED** [review]
Minor comment on style. Otherwise, LGTM

File: idaes/core/util/misc.py (hunk 3 above)
Comment (andrewlee94) [reply idaes/core/util/misc.py:225]: If `b` is not a block, you need to read the documentation better. On the subject of `b` versus `block`, I think that should be part of the broader discussion @lbianchi-lbl started to standardize everything.

Comment (jghouse88) [reply idaes/core/util/misc.py:225]: > 
> 
> If `b` is not a block, you need to read the documentation better. On the subject of `b` versus `block`, I think that should be part of the broader discussion @lbianchi-lbl started to standardize everything.

Assuming someone constantly looks at the documentation. ;)

**Pull request merged**